src/Rpc/NodeGuardService.cs

```diff
@@ -748,4 +786,99 @@ private bool ValidateTargetBalance(LiquidityRule rule)
 
         return true;
     }
+
+    public override async Task<GetAvailableUtxosResponse> GetAvailableUtxos(GetAvailableUtxosRequest request, ServerCallContext context)
+    {
+        var wallet = await _walletRepository.GetById(request.WalletId);
+        if (wallet == null)
+        {
+            throw new Exception("Wallet not found");
+        }
+
+        CoinSelectionStrategy coinSelectionStrategy;
+        switch (request.Strategy)
+        {
+            case COIN_SELECTION_STRATEGY.BiggestFirst:
+                coinSelectionStrategy = CoinSelectionStrategy.BiggestFirst;
+                break;
+            case COIN_SELECTION_STRATEGY.SmallestFirst:
+                coinSelectionStrategy = CoinSelectionStrategy.SmallestFirst;
+                break;
+            case COIN_SELECTION_STRATEGY.ClosestToTargetFirst:
+                coinSelectionStrategy = CoinSelectionStrategy.ClosestToTargetFirst;
+                break;
+            case COIN_SELECTION_STRATEGY.UpToAmount:
+                coinSelectionStrategy = CoinSelectionStrategy.UpToAmount;
+                break;
+            default:
+                coinSelectionStrategy = CoinSelectionStrategy.SmallestFirst;
+                break;
+        }
```

Comment:
but if we use casts and someone adds a new item in the middle of the `CoinSelectionStrategy`, then the cast is broken. But I will convert this to pattern match (since I recently discovered the syntax 😛) and throw if there's no recongnized selection
```
var coinSelectionStrategy = request.Strategy switch {
    COIN_SELECTION_STRATEGY.BiggestFirst => CoinSelectionStrategy.BiggestFirst;
    ....
    ....
    _ => throw new ArgumentOutOfRange()
}
```